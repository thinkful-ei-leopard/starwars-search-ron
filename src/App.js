import React from 'react';
import './App.css';

// Users should be able to search for a character name
// App will make an API request to the Star Wars API
// with the response, render it out to the user
// display Matching Characters for their search. 
//    ex. search "Skywalker"
//    should RETURN "Luke Skywalker", "Anakin Skywalker", "Shmi Skywalker", etc
// --MUST deploy to Zeits Now

// BONUS:
// -Gracefully displaying a loading indicator 
//    and handling error conditions (validation, error rendering)
// -Be able to SELECT whether they want to search for planets, spaceships, vehicles, characters,
//    films, or species
// - A passing test suite

class App extends React.Component {

  // For Fetch. Search people by this: where r2 is the string you wish to search.
  // https://swapi.co/api/people/?search=r2

  state = {};

  componentDidMount() {

  }

  render() {
    return (
      <div className="App">
        <h1>Test</h1>
      </div>
    );
  }
}

export default App;
