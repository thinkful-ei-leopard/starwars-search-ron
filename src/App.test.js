import React from 'react';
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';
import App from './App';

describe('<App />', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<App />, div);
        ReactDOM.unmountComponentAtNode(div);
    });
    it('renders this UI as expected', () => {
        const tree= renderer.create(<App />).toJSON();
        expect(tree).toMatchSnapshot();
    });    
});