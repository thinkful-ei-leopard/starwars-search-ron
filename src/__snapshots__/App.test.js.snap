// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders this UI as expected 1`] = `
<div
  className="App"
>
  <h1>
    Star Wars Search App!
  </h1>
  <h3>
    A long time ago, in an API far, far away...
  </h3>
  <form
    className="Form__container"
    onSubmit={[Function]}
  >
    <div
      className="form-group"
    >
      <label
        htmlFor="search-character-name"
      >
        Search characters name: 
      </label>
      <input
        className="Form__control"
        id="search-character-name"
        name="search-character-name"
        onChange={[Function]}
        required={true}
        type="text"
      />
    </div>
    <div
      className="Form__button__group"
    >
      <button
        className="Form__button"
        disabled="Please enter search term"
        type="submit"
      >
        Search
      </button>
    </div>
  </form>
  <ul
    className="Results__container"
  />
</div>
`;
